tools/clang/test/HLSL/rewriter/correct_rewrites/cpp-errors_gold.hlsl

```diff
@@ -105,7 +105,7 @@ namespace ns_with_struct {
     int i;
   };
 }
-matrix<int, 1, 2> g_matrix_simple;
+const matrix<int, 1, 2> g_matrix_simple;
```

Comment:
Should const be getting added here?

Reply:
We mark const on global variables which will be in global cbuffer
But template matrix type is not canonical so it will not be marked while ast matrix type is canonical.